cli/cmd/root.go

```diff
@@ -126,31 +127,51 @@ func runCmd(ctx context.Context, ver cmdutil.Version) error {
 	rootCmd.PersistentFlags().StringVar(&ch.AdminTokenOverride, "api-token", "", "Token for authenticating with the cloud API")
 	rootCmd.Flags().BoolP("version", "v", false, "Show rill version") // Adds option to get version by passing --version or -v
 
-	// Add sub-commands
-	rootCmd.AddCommand(
+	// Command Groups
+
+	// Project commands
+	cmdutil.AddGroup(rootCmd, "Project",
 		start.StartCmd(ch),
 		deploy.DeployCmd(ch),
+		project.ProjectCmd(ch),
+		publicurl.PublicURLCmd(ch),
 		env.EnvCmd(ch),
+	)
+
+	// Organization commands
+	cmdutil.AddGroup(rootCmd, "Organization",
+		org.OrgCmd(ch),
 		user.UserCmd(ch),
 		usergroup.UsergroupCmd(ch),
-		org.OrgCmd(ch),
-		project.ProjectCmd(ch),
-		publicurl.PublicURLCmd(ch),
 		service.ServiceCmd(ch),
+		billing.BillingCmd(ch),
+	)
+
+	// Auth commands
+	cmdutil.AddGroup(rootCmd, "Auth",
 		auth.LoginCmd(ch),
 		auth.LogoutCmd(ch),
 		whoami.WhoamiCmd(ch),
-		docs.DocsCmd(ch, rootCmd),
+	)
+
+	if ch.IsDev() {
+		// Internal commands - hidden from help menu
+		cmdutil.AddGroup(rootCmd, "Internal",
```

Comment:
This will cause the internal commands to not be added if `!ch.IsDev()` – we still need them to be added, but just marked as hidden so they don't show up in the root help menu (you can think of the internal commands as undocumented APIs). For example, if you grab the latest release, running `rill -h` does not describe the `sudo` command, but you can still run `rill sudo -h` to access the sudo command if you know it's there.

Reply:
Ahh that makes sense, swapping that out